streaming/stream_request.hh

```diff
@@ -29,12 +29,12 @@ public:
     }
     std::vector<sstring> column_families;
     stream_request() = default;
-    stream_request(sstring _keyspace, dht::token_range_vector _ranges, std::vector<sstring> _column_families)
+    stream_request(sstring _keyspace, dht::token_range_vector&& _ranges, std::vector<sstring>&& _column_families) noexcept
```

Comment:
Re commit message: `std::vector` move constructor is `noexcept` since C++17 so the `stream_request` ctor could already have been made `noexcept` earlier.

Reply:
And you're invoking `std::vector` move constructor below anyway (`ranges(std::move(_ranges))` etc.), so you are depending on the `noexcept`ness of `std::vector`'s move constructor anyway if you want to mark this ctor `noexcept`.

Reply:
true